File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_Audio/CM_Audio WatchKit Extension/InterfaceController.m
//
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "InterfaceController.h"

@interface InterfaceController()

@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
}

- (void)willActivate {
    [super willActivate];
}

- (void)didDeactivate {
    [super didDeactivate];
}

#pragma mark - Private Methods

- (IBAction)actionRecorder {
    NSURL *container = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.cm_audio"];
    NSURL *fileURL = [container URLByAppendingPathComponent:@"audio.mp4"];
    
    NSDictionary *options = @{ WKAudioRecorderControllerOptionsMaximumDurationKey : @(10),
                               WKAudioRecorderControllerOptionsActionTitleKey : @"Test" };
    
    [self presentAudioRecorderControllerWithOutputURL:fileURL preset:WKAudioRecorderPresetNarrowBandSpeech options:options completion:^(BOOL didSave, NSError * _Nullable error) {
        // NOP
    }];
}

- (IBAction)actionPlay {
    NSURL *container = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:@"group.cm_audio"];
    NSURL *fileURL = [container URLByAppendingPathComponent:@"audio.mp4"];
    
    [self presentMediaPlayerControllerWithURL:fileURL options:nil completion:^(BOOL didPlayToEnd, NSTimeInterval endTime, NSError * _Nullable error) {
        // NOP
    }];
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_Audio/CM_Audio WatchKit Extension/InterfaceController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import WatchKit
import Foundation

class InterfaceController: WKInterfaceController {
    
    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
    }
    
    override func willActivate() {
        super.willActivate()
    }
    
    override func didDeactivate() {
        super.didDeactivate()
    }
    
    // MARK: - Private Methods
    
    @IBAction func actionRecorder() {
        let container = NSFileManager.defaultManager().containerURLForSecurityApplicationGroupIdentifier("group.cm_audio")
        let fileURL = container!.URLByAppendingPathComponent("audio.mp4")
        let options = [
            WKAudioRecorderControllerOptionsMaximumDurationKey : 10,
            WKAudioRecorderControllerOptionsActionTitleKey : "Test"
        ]
        
        self.presentAudioRecorderControllerWithOutputURL(fileURL, preset: .NarrowBandSpeech, options: options) { (value: Bool, error: NSError?) -> Void in
            // NOP
        }
    }
    
    @IBAction func actionPlay() {
        let container = NSFileManager.defaultManager().containerURLForSecurityApplicationGroupIdentifier("group.cm_audio")
        let fileURL = container!.URLByAppendingPathComponent("audio.mp4")
        
        self.presentMediaPlayerControllerWithURL(fileURL, options: nil) { (value: Bool, time: NSTimeInterval, error: NSError?) -> Void in
            // NOP
        }
    }
}
